opensrp-opd/src/main/java/org/smartregister/opd/utils/OpdUtils.java

```diff
@@ -371,4 +374,39 @@ public static JSONArray addOpenMrsEntityId(@NonNull String diagnosisType, @NonNu
         }
         return jsonArray;
     }
+
+    @NonNull
+    public static JSONArray generateFieldsFromJsonForm(@NonNull JSONObject jsonFormObject) throws JSONException {
```

Comment:
I think we have this functionality in JsonFormUtils#fields method of native form library

Reply:
ok changed